ethereum/contracts/SwapCreator.sol

```diff
@@ -1,61 +1,88 @@
+    // by Alice calling `refund`.
     enum Stage {
         INVALID,
         PENDING,
         READY,
         COMPLETED
     }
 
+    // swaps maps from a swap ID to the swap's current Stage
+    mapping(bytes32 => Stage) public swaps;
+
+    // Swap stores the initial swap configuration, the hash of which forms the
+    // swap ID.
     struct Swap {
-        // the swap initiator, Alice
-        // address allowed to refund the ether for this swap
+        // owner is the address of Alice, who initiates the swap by calling
+        // `newSwap`. Only the owner is allowed to call `setReady` or `refund`.
         address payable owner;
-        // address allowed to claim the ether for this swap, Bob
+        // claimer is the address of Bob. Only the claimer can call `claim` or
+        // sign a RelaySwap object that `claimRelayer` will accept the signature
+        // for.
         address payable claimer;
-        // the keccak256 hash of the expected public key derived from the secret `s_b`.
-        // this public key is a point on the secp256k1 curve
+        // pubKeyClaim is the Keccak-256 hash of the expected public key derived
+        // from the secret that Bob sends when claiming. This public key is a
+        // point on the secp256k1 curve. Bob gives Alice this public key hash
+        // and proves that he already knows the secret associated with it, and
+        // that the secret also forms a valid point on the ed25519 curve,
+        // off-chain before Alice initiates the swap on-chain.
         bytes32 pubKeyClaim;
-        // the keccak256 hash of the expected public key derived from the secret `s_a`.
-        // this public key is a point on the secp256k1 curve
+        // pubKeyRefund is the Keccak-256 hash of the expected public key
+        // derived from the secret Alice sends if refunding. In successful
+        // swaps, Alice never reveals the secret. This public key is a point on
+        // the secp256k1 curve. Alice proves to Bob that she already knows the
+        // secret associated with this public key, and that the secret also
+        // forms a valid point on the ed25519 curve, off-chain before starting
+        // the swap on-chain.
         bytes32 pubKeyRefund;
-        // timestamp before which Alice can call either `setReady` or `refund`
+        // timeout1 is the block timestamp before which Alice can call
+        // either `setReady` or `refund`.
         uint256 timeout1;
-        // timestamp after which Bob cannot claim, only Alice can refund
+        // timeout2 is the block timestamp after which Bob cannot claim, and
+        // after which Alice can refund if Bob did not claim.
         uint256 timeout2;
-        // the asset being swapped: equal to address(0) for ETH, or an ERC-20 token address
+        // asset is address(0) for ETH swaps, or an ERC-20 address if Alice is
+        // providing tokens for XMR.
```

Comment:
I think this was clearer before

Reply:
Original:
```text
        // the asset being swapped: equal to address(0) for ETH, or an ERC-20 token address
        address asset;
```
Latest version:
```text
        // asset is address(0) for EVM native currency swaps, or it is the
        // address of the token that Alice is providing.
        address asset;
```
At the top of the contract, we clarify our definition of the word "token" for rest of the documentation. It could be a BEP-20 token. The field name was shifted to the first word for style consistency.